BootloaderCorePkg/Stage2/Stage2Hob.c

```diff
@@ -605,6 +605,164 @@ BuildUniversalPayloadHob (
 }
 
 
+/**
+  Build SMM variable related HOBs for Universal Payload
+
+**/
+VOID
+BuildSmmVariableHobs (
+  VOID
+)
+{
+  EFI_STATUS                           Status;
+  LDR_SMM_INFO                         *SmmInfoHob;
+  UINT32                               Length;
+  PLD_SMM_REGISTERS                    *SmmRegisterHob;
+  PLD_GENERIC_REGISTER                 *Reg;
+  EFI_SMRAM_HOB_DESCRIPTOR_BLOCK       *SmmMemoryHob;
+  SPI_FLASH_INFO                       *SpiFlashInfoHob;
+  LOADER_PLATFORM_INFO                 *PlatformInfo;
+  NV_VARIABLE_INFO                     *NvVariableHob;
+  UINT32                               Base;
+  PLD_S3_COMMUNICATION                 *S3CommunicationHob;
+
+  if ((PcdGet8(PcdBuildSmmHobs) & BIT0) != 0) {
+    SmmInfoHob = (LDR_SMM_INFO *)GetGuidHobData (NULL, NULL, &gSmmInformationGuid);

```

Comment:
Add SmmInfoHob = NULL at beginning to ensure it is initialized.

Reply:
I will add it. But it is not necessary since SmmInfoHob will be initialized by line 630 when BIT0 is set in PCD PcdBuildSmmHobs and SmmInfoHob is only used when BIT0 is set in PCD PcdBuildSmmHobs.